test/e2e/platform/platform_test.go

```diff
@@ -37,65 +39,248 @@ import (
 	"tkestack.io/tke/test/util/cloudprovider/tencent"
 )
 
-var t = tke.TKE{Namespace: "platform"}
+var (
+	t                 = tke.TKE{Namespace: "platform"}
+	tkeClient         *tkeclientset.Clientset
+	provider          = tencent.NewTencentProvider()
+	err               error
+	tkeKubeConfigFile string
+	clusterNames      []string
+)
 
 var _ = BeforeSuite(func() {
 	t.Create()
+
+	tkeKubeConfigFile = t.GetKubeConfigFile()
+	restConf, err := t.GetKubeConfig()
+	Expect(err).To(BeNil())
+	tkeClient = tkeclientset.NewForConfigOrDie(restConf)
 })
 
 var _ = AfterSuite(func() {
+	for _, name := range clusterNames {
+		Expect(deleteCluster(name)).Should(Succeed())
+	}
+
 	t.Delete()
+
+	if os.Getenv("NEED_DELETE") == "" {
+		return
+	}
+	Expect(provider.DeleteAllInstances()).Should(Succeed())
 })
 
-var _ = Describe("cluster lifecycle", func() {
-	var provider cloudprovider.Provider
+var _ = Describe("Platform Test", func() {
 
-	var masterNodes []cloudprovider.Instance
-	var workerNodes []cloudprovider.Instance
+	Context("Baremetal cluster", func() {
+		var clsName string
 
-	BeforeEach(func() {
-		var err error
-		provider = tencent.NewTencentProvider()
-		masterNodes, err = provider.CreateInstances(2)
-		Expect(err).To(BeNil())
-		workerNodes, err = provider.CreateInstances(1)
-		Expect(err).To(BeNil())
-	})
+		BeforeEach(func() {
+			// Create Baremetal cluster
+			if clsName == "" {
+				cls, err := createCluster(1)
+				Expect(err).To(BeNil())
+				clsName = cls.Name
+			}
+		})
 
-	AfterEach(func() {
-		if os.Getenv("NEED_DELETE") == "" {
-			return
-		}
-		var instanceIDs []*string
-		for i, one := range masterNodes {
-			fmt.Printf("delete instance %d %s\n", i, one.InternalIP)
-			instanceIDs = append(instanceIDs, &masterNodes[i].InstanceID)
-		}
-		for i, one := range workerNodes {
-			fmt.Printf("delete instance %d %s\n", i, one.InternalIP)
-			instanceIDs = append(instanceIDs, &workerNodes[i].InstanceID)
-		}
-		err := provider.DeleteInstances(instanceIDs)
-		Expect(err).To(BeNil())
+		It("Create Baremetal cluster", func() {
+			out, err := runCmd("kubectl get clusters --kubeconfig " + tkeKubeConfigFile + " | grep " + clsName)
+			Expect(err).Should(BeNil())
```

Comment:
if you depends on `kubectl`  to create cluster, do you mean we need install kubectl on the CI host? how to handle kubectl upgrade?  why node use go api to do this?

Reply:
It doesn't depend on kubectl to create cluster. kubectl was introduced to do some check as mentioned in the manual test cases.

Reply:
why need `kubectl`, who install it on CI host? why not use go-clinet/go-api to do check?

Reply:
As discussed offline, I'll remove `kubectl` related code, but make sure a kubeconfig file generated there for easily use if needed.